packages/web-components/src/components/masthead/masthead-global-bar.ts

```diff
@@ -21,16 +22,70 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-masthead-global-bar`)
 class DDSMastheadGlobalBar extends FocusMixin(LitElement) {
+  /**
+   * Media query on medium breakpoint.
+   */
+  private mqMedium;
+
   /**
    * The shadow slot this action bar should be in.
    */
   @property({ reflect: true })
   slot = 'profile';
 
+  /**
+   * Search bar opened flag.
+   */
+  @internalProperty()
+  private openedSearch = false;
+
+  /**
+   * Medium breakpoint flag.
+   */
+  @internalProperty()
+  private responsiveMd = false;
+
+  /**
+   * Handles toggle event from the search component.
+   *
+   * @param event The event.
+   */
+  private _handleSearchToggle(event: Event) {
+    this.openedSearch = (event as CustomEvent).detail.active;
+  }
+
+  /**
+   * Handles media query event.
+   *
+   * @param event The event.
+   */
+  private _handleMediaQuery(event: Event) {
+    console.log(event);
+    this.responsiveMd = (event as MediaQueryListEvent).matches;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.getRootNode().addEventListener('dds-masthead-search-toggled', this._handleSearchToggle.bind(this));
```

Comment:
We make these custom event names configurable. Please see https://github.com/carbon-design-system/ibm-dotcom-library/blob/master/packages/web-components/src/components/footer/footer-composite.ts#L139-L144 for more details.